packages/legend-query/src/stores/QueryBuilderPropertySearchPanelState.ts

```diff
@@ -0,0 +1,310 @@
+        node.type.name === PRIMITIVE_TYPE.STRING
+      ) {
+        return true;
+      }
+      return false;
+    });
+  }
+
+  numberPropertyNodes(): QueryBuilderExplorerTreeNodeData[] {
+    return this.searchedMappedPropertyNodes.filter((node) => {
+      if (
+        node.type instanceof PrimitiveType &&
+        (node.type.name === PRIMITIVE_TYPE.DECIMAL ||
+          node.type.name === PRIMITIVE_TYPE.NUMBER ||
+          node.type.name === PRIMITIVE_TYPE.INTEGER ||
+          node.type.name === PRIMITIVE_TYPE.FLOAT)
+      ) {
+        return true;
+      }
+      return false;
+    });
+  }
+
+  datePropertyNodes(): QueryBuilderExplorerTreeNodeData[] {
+    return this.searchedMappedPropertyNodes.filter((node) => {
+      if (
+        node.type instanceof PrimitiveType &&
+        (node.type.name === PRIMITIVE_TYPE.DATE ||
+          node.type.name === PRIMITIVE_TYPE.DATETIME ||
+          node.type.name === PRIMITIVE_TYPE.STRICTDATE ||
+          node.type.name === PRIMITIVE_TYPE.STRICTTIME ||
+          node.type.name === PRIMITIVE_TYPE.LATESTDATE)
+      ) {
+        return true;
+      }
+      return false;
+    });
+  }
+
+  booleanPropertyNodes(): QueryBuilderExplorerTreeNodeData[] {
+    return this.searchedMappedPropertyNodes.filter((node) => {
+      if (
+        node.type instanceof PrimitiveType &&
+        node.type.name === PRIMITIVE_TYPE.BOOLEAN
+      ) {
+        return true;
+      }
+      return false;
+    });
+  }
+
+  get filteredPropertyNodes(): QueryBuilderExplorerTreeNodeData[] {
+    return this.searchedMappedPropertyNodes.filter((p) => {
+      if (
+        !this.isOneManyRowsIncluded &&
+        this.oneManyPropertyNodes().includes(p)
+      ) {
+        return false;
+      }
+      if (!this.isClassIncluded && this.classPropertyNodes().includes(p)) {
+        return false;
+      }
+      if (!this.isStringIncluded && this.stringPropertyNodes().includes(p)) {
+        return false;
+      }
+      if (!this.isNumberIncluded && this.numberPropertyNodes().includes(p)) {
+        return false;
+      }
+      if (!this.isBooleanIncluded && this.booleanPropertyNodes().includes(p)) {
+        return false;
+      }
+      if (!this.isDateIncluded && this.datePropertyNodes().includes(p)) {
+        return false;
+      }
+      return true;
+    });
+  }
+
+  fetchAllPropertyNodes(): void {
+    const treeData = this.queryBuilderState.explorerState.treeData;
+    let currentLevelPropertyNodes: QueryBuilderExplorerTreeNodeData[] = [];
+    let nextLevelPropertyNodes: QueryBuilderExplorerTreeNodeData[] = [];
+    if (treeData?.nodes.values()) {
+      Array.from(treeData.nodes.values())
+        .slice(1)
+        .forEach((node) => {
+          if (node.mappingData.mapped && !node.isPartOfDerivedPropertyBranch) {
+            currentLevelPropertyNodes.push(node);
+            this.allMappedPropertyNodes.push(node);
+          }
+        });
+    }
+    let currentDepth = 1;
+    const maxDepth = 15;
```

Comment:
let's make this a constant and put it in `QueryBuilder_Const`: `QUERY_BUILDER_PROPERTY_SEARCH_MAX_DEPTH = 15`